app/actions/views/channel.js

```diff
@@ -18,18 +22,66 @@ export function loadChannelsIfNecessary(teamId) {
             }
         }
 
-        if (!hasChannelsForTeam) {
+        if (hasChannelsForTeam) {
+            await getMyChannelMembers(teamId)(dispatch, getState);
+        } else {
             await fetchMyChannelsAndMembers(teamId)(dispatch, getState);
         }
     };
 }
 
+export function loadProfilesAndTeamMembersForDMSidebar(teamId) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {currentId: currentUserId} = state.entities.users;
```

Comment:
Can we keep it so that we only use destructuring when the local variable name matches the field? I have trouble remembering exactly what the variable ends up being called with this syntax.